### PR TITLE
PRM-448 Updated CYA page and added makingALateAppeal.bereavementReaso…

### PR DESCRIPTION
**Pull request by ruthdas03**
…n.error.required

### CODE REVIEW

File: test/controllers/MakingALateAppealControllerSpec.scala [hunk 1]
```diff
@@ -47,12 +47,14 @@ class MakingALateAppealControllerSpec extends SpecBase {
...
-        val result = controller.onPageLoad()(fakeRequestWithCorrectKeys)
+        val result = controller.onPageLoad()(fakeRequestWithCorrectKeys.withSession(SessionKeys.reasonableExcuse -> "crime"))
         status(result) shouldBe OK
       }
 
       "return OK and correct view (pre-populated textbox when present in session)" in new Setup(AuthTestModels.successfulAuthResult) {
-        val result = controller.onPageLoad()(fakeRequestConverter(fakeRequestWithCorrectKeys.withSession(SessionKeys.lateAppealReason -> "This is a reason.")))
+        val result = controller.onPageLoad()(fakeRequestConverter(fakeRequestWithCorrectKeys
+          .withSession(SessionKeys.lateAppealReason -> "This is a reason.")
+        .withSession(SessionKeys.reasonableExcuse -> "crime")))
```
Comment (LewisMitchell) [inline test/controllers/MakingALateAppealControllerSpec.scala:57]: Same as above

File: test/views/MakingALateAppealPageSpec.scala [hunk 2]
```diff
@@ -33,7 +34,8 @@ class MakingALateAppealPageSpec extends SpecBase with ViewBehaviours {
     }
     def applyView(form: Form[_]): HtmlFormat.Appendable = makingALateAppealPage.apply(form)
 
-    val formProvider = MakingALateAppealForm.makingALateAppealForm
+    val formProvider = MakingALateAppealForm.makingALateAppealForm(fakeRequest.
+      withSession(SessionKeys.reasonableExcuse -> "crime"))
```
Comment (LewisMitchell) [inline test/views/MakingALateAppealPageSpec.scala:38]: Should be able to revert this if the custom error message is not needed

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Looks good, just some minor changes. AT's pass.

File: conf/messages [hunk 3]
```diff
@@ -335,6 +337,7 @@ makingALateAppeal.title = Making a late appeal
 makingALateAppeal.heading = Making a late appeal
 makingALateAppeal.p1 = You usually need to appeal within 30 days of being told about a penalty. If you do not appeal within 30 days, we’ll use information you provide here to decide whether we can consider your late appeal.
 makingALateAppeal.error.required = You must provide some information about why you did not appeal sooner
+makingALateAppeal.bereavementReason.error.required = Explain why you did not appeal sooner
```
[the remark this answers is not in the document]
Comment (ruthdas03) [reply conf/messages]: Yes, removed the same.

File: app/forms/MakingALateAppealForm.scala [hunk 4]
```diff
@@ -19,12 +19,20 @@ package forms
...
+  def makingALateAppealForm(implicit request: Request[_]) = Form[String](
     single(
-      "late-appeal-text" -> text(errorKey = "makingALateAppeal.error.required")
+      if(request.session.get(SessionKeys.reasonableExcuse)
+        .get.equalsIgnoreCase("bereavement"))
+        "late-appeal-text" ->
+          text(errorKey = "makingALateAppeal.bereavementReason.error.required")
+    else
+      "late-appeal-text" ->
+        text(errorKey = "makingALateAppeal.error.required")
```
[the remark this answers is not in the document]
Comment (ruthdas03) [reply app/forms/MakingALateAppealForm.scala]: Removed the error and reverted the logic back

File: app/controllers/MakingALateAppealController.scala [hunk 5]
```diff
@@ -43,7 +43,7 @@ class MakingALateAppealController @Inject()(makingALateAppealPage: MakingALateAp
 
   def onSubmit(): Action[AnyContent] = (authorise andThen dataRequired) {
     implicit request => {
-      MakingALateAppealForm.makingALateAppealForm().bindFromRequest().fold(
+      MakingALateAppealForm.makingALateAppealForm(request).bindFromRequest().fold(
```
[the remark this answers is not in the document]
Comment (ruthdas03) [reply app/controllers/MakingALateAppealController.scala]: Done

File: conf/messages.cy [hunk 6]
```diff
@@ -334,6 +336,7 @@ makingALateAppeal.title = Making a late appeal
 makingALateAppeal.heading = Making a late appeal
 makingALateAppeal.p1 = You usually need to appeal within 30 days of being told about a penalty. If you do not appeal within 30 days, we’ll use information you provide here to decide whether we can consider your late appeal.
 makingALateAppeal.error.required = You must provide some information about why you did not appeal sooner
+makingALateAppeal.bereavementReason.error.required = Explain why you did not appeal sooner
```
[the remark this answers is not in the document]
Comment (ruthdas03) [reply conf/messages.cy]: Removed it.

File: test/controllers/MakingALateAppealControllerSpec.scala [hunk 7]
```diff
@@ -47,12 +47,14 @@ class MakingALateAppealControllerSpec extends SpecBase {
   "onPageLoad" should {
     "return 200" when {
       "the user is authorised and has the correct keys in the session" in new Setup(AuthTestModels.successfulAuthResult) {
-        val result = controller.onPageLoad()(fakeRequestWithCorrectKeys)
+        val result = controller.onPageLoad()(fakeRequestWithCorrectKeys.withSession(SessionKeys.reasonableExcuse -> "crime"))
```
[the remark this answers is not in the document]
Comment (ruthdas03) [reply test/controllers/MakingALateAppealControllerSpec.scala]: For now removed the reasonable excuse.

File: test/controllers/MakingALateAppealControllerSpec.scala (hunk 1 above)
Comment (ruthdas03) [reply test/controllers/MakingALateAppealControllerSpec.scala]: Done.

File: test/views/MakingALateAppealPageSpec.scala (hunk 2 above)
Comment (ruthdas03) [reply test/views/MakingALateAppealPageSpec.scala]: Yes custom message not needed, hence removed.

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Could you add some more tests for the CheckYourAnswersController?
One for unit tests
https://github.com/hmrc/penalties-appeals-frontend/blob/0b0c8ef6207ff03db4f9d7e2e5925f4f20df5cc5/test/controllers/CheckYourAnswersControllerSpec.scala#L126-L129
One for integration tests
https://github.com/hmrc/penalties-appeals-frontend/blob/0b0c8ef6207ff03db4f9d7e2e5925f4f20df5cc5/it/controllers/CheckYourAnswersControllerISpec.scala#L31-L51

**Comment by LewisMitchell** [comment]
Other than the above, looks good

**Comment by LewisMitchell** [comment]
image

**Comment by ruthdas03** [comment]
> Other than the above, looks good

Added the test cases

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**